Thoughtless-Eels/Bangazon-WebApp
# Cas view product categories

## Link to Ticket
Ticket: # [Link]()

## Description of Proposed Changes
Given a user is viewing any page on the Bangazon site
When the user performs a gesture on the Product Categories hyperlink
Sees the  product categories you have submitted
And next to the category name, the number of products in that category will be displayed
And the first three products in the category will be displayed beneath each category
- 

## Steps to Test
Go to The Products Categories Page
You should see the categories of any products that you have already seeded 
You should also see the number of products in each category displaying to the left of the CategoryName
Should also see the first three products created in each category

Outline the steps to test

```sh
git fetch --all
git checkout CAS-ViewProductCategories

```


## Impacted Areas in Application
The Product Type Controller, Model, View Model
List general components of the application that this PR will affect:

* 

## Mentions @Gward2489 
We paired up and worked on this ticket for the majority of the afternoon yesterday. Teamwork, 💯 .


Tag users that need to review this code

## Definition of Done

1. The project must be fully documented. This includes the following:
    1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
    1. Every class must be documented with purpose, author, and methods.
    1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each.
